### PR TITLE
Use AttestationEvidence for evidence coming from the enclave

### PR DESCRIPTION
**Pull request by rbehjati**
Some renaming and refactoring based on a chat with @ipetr0v to make the terminology more consistent with RATS.

### CODE REVIEW

File: oak_remote_attestation_noninteractive/proto/v1/messages.proto [hunk 1]
```diff
@@ -21,9 +21,7 @@ package oak.session.noninteractive.v1;
 option java_multiple_files = true;
 option java_package = "oak.session.noninteractive.v1";
 
-// AttestationEvidence contains the information that the untrusted launcher provides to the client
-// in response to its request for the enclave's public key(s). The name is chosen to match the RATS
-// terminology (see https://www.rfc-editor.org/rfc/rfc9334.html#name-evidence).
+// AttestationEvidence contains the evidence provided by the enclave.
 message AttestationEvidence {
```
Comment (ipetr0v) [inline oak_remote_attestation_noninteractive/proto/v1/messages.proto:25]: I think this one is the Evidence from RATS: https://www.rfc-editor.org/rfc/rfc9334.html#name-evidence (could you please add this link to the comments) ?

File: oak_remote_attestation_noninteractive/proto/v1/messages.proto [hunk 2]
```diff
@@ -37,22 +35,30 @@ message AttestationEvidence {
...
+  // The signature over the application_data, signed by the enclave using its signing key. This
+  // field is optional, and must be left empty if application_data is not provided.
+  // TODO(#3442): Specify the signing details, or link to a specification.
+  bytes signed_application_data = 4;
+}
+
+// AttestationVerificationEvidence contains the information that the untrusted launcher provides to
+// the client in response to its request for the enclave's public key(s). The name is chosen to
+// match the RATS terminology (see https://www.rfc-editor.org/rfc/rfc9334.html#name-evidence).
+message AttestationVerificationEvidence {
```
[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_remote_attestation_noninteractive/proto/v1/messages.proto:47]: > And this one is the RATS endorsement: https://datatracker.ietf.org/doc/html/rfc9334#name-endorsements (could you please also add this link to the comments)

 I am not sure about this. From the spec, it does not sound like that `RATS Endorsement` contains `RATS evidence`, whereas `AttestationVerificationEvidence` contains `AttestationEvidence`. 

@conradgrobler @tiziano88 What do you think?

> I mean, we can split the AttestationEvidence and the AttestationEndorsement into separate messages, and then have a combined message that just has 2 fields.
> 
> And the combined message could be called `OakEvidence`

What is `AttestationEndorsement`?

How can the combination of a bunch of RATS things become an Oak thing? Are we the only ones using Evidence and Endorsement together like this? Why does not RATS have a name for the combination of `Evidence` and `Endorsement`?

[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_remote_attestation_noninteractive/proto/v1/messages.proto:47]: Based on my understanding of https://datatracker.ietf.org/doc/html/rfc9334#name-endorsements, the only part of `AttestationVerificationEvidence`  that qualifies as a RATS Endorsement is `tee_certificates.` Everything else in it qualifies as _Claims about the Target Environment that reveal operational status, health, configuration, or construction that have security relevance_. And a `RATS Evidence` is a set of such claims. So I think the comments are correct.

[the remark this answers is not in the document]
Comment (ipetr0v) [reply oak_remote_attestation_noninteractive/proto/v1/messages.proto:47]: We also could just call the top-level message `Attestation`.

Which will include `Evidence`, `Endorsement` and any additional data.

File: oak_remote_attestation_noninteractive/proto/v1/messages.proto (hunk 1 above)
Comment (rbehjati) [reply oak_remote_attestation_noninteractive/proto/v1/messages.proto:27]: Done.

File: oak_remote_attestation_noninteractive/proto/v1/messages.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_remote_attestation_noninteractive/proto/v1/messages.proto]: Thanks for comments. @conradgrobler's description matches my understanding too. 

I've changed `AttestationVerificationEvidence` to `AttestationBundle`. It now has two fields: `AttestationEvidence` and `AttestationEndorsement`. 
I prefer these prefixed named over just `Attestation`,  `Evidence`, `Endorsement`, as the prefixed ones are clearer and less confusing.

**Pull request merged**